File: account-service/src/main/java/com/food/delivery/accountservice/company/CompanyClientFallbackFactory.java
package com.food.delivery.accountservice.company;

import lombok.extern.slf4j.Slf4j;
import reactivefeign.FallbackFactory;
import reactor.core.publisher.Mono;

@Slf4j
class CompanyClientFallbackFactory implements FallbackFactory<CompanyClient> {
	@Override
	public CompanyClient apply(Throwable throwable) {
		return companyId -> {
			log.warn("Company with id: {} not found", companyId);
			return Mono.empty();
		};
	}
}

File: menu-service/src/main/kotlin/com/food/delivery/menuservice/company/CompanyClientFallbackFactory.kt
package com.food.delivery.menuservice.company

import org.slf4j.Logger
import org.slf4j.LoggerFactory
import reactivefeign.FallbackFactory
import reactor.core.publisher.Mono

internal class CompanyClientFallbackFactory : FallbackFactory<CompanyClient> {

    val logger: Logger = LoggerFactory.getLogger(CompanyClientFallbackFactory::class.java)

    override fun apply(throwable: Throwable): CompanyClient {
        return object : CompanyClient {
            override fun getCompany(companyId: String): Mono<CompanyRest> {
                logger.error("No company found with given companyId: $companyId")
                return Mono.empty<CompanyRest>()
            }
        }
    }
}
